packages/legacy/core/App/screens/Splash.tsx

```diff
@@ -230,6 +230,32 @@ const Splash: React.FC = () => {
           return
         }
 
+        if (agent) {
+          logger.info('Agent already initialized, restarting...')
+
+          try {
+            await agent.wallet.open({
+              id: walletSecret.id,
+              key: walletSecret.key,
+            })
+          } catch (error: unknown) {
+            logger.error('Error opening existing wallet', error as Error)
+          }
+
+          await agent.mediationRecipient.initiateMessagePickup()
+
+          navigation.dispatch(
+            CommonActions.reset({
+              index: 0,
+              routes: [{ name: Stacks.TabStack }],
+            })
+          )
+
+          return
+        }
+
+        logger.info('No agent initialized, creating a new one')
+
         await (ocaBundleResolver as RemoteOCABundleResolver).checkForUpdates?.()
```

Comment:
Could we move this above the early return for existing agents so that we still check for OCA updates?